src/components/ModelizerTextView.vue

```diff
@@ -67,101 +62,136 @@ const props = defineProps({
-  nodes.value = getTree(props.projectName, fileInformations);
-  fileTabArray.value = fileTabArray.value.filter(({ id }) => projectFilesIds.includes(id));
-
-  fileTabArray.value.forEach((file) => {
-    readProjectFile(props.projectName, { path: file.id })
-      .then(({ content }) => {
-        file.content = content;
-      });
-  });
+async function onCommitFiles(committedFiles) {
+  const filePaths = committedFiles.map(({ path }) => path);
 
-  const isActiveFileInFiles = fileTabArray.value
-    .find(({ id }) => id === activeFileTab.value.id);
+  const fileStatus = await getStatus(
+    props.projectName,
+    filePaths,
+    (path) => filePaths.includes(path),
+  );
 
-  if (!isActiveFileInFiles) {
-    setLastFileActive();
+  localFileInformations.value = localFileInformations.value
+    .map((fileInformation) => fileStatus.find(({ path }) => path === fileInformation.path)
+      || fileInformation);
+}
+
+/**
+ * Add created file to localFileInformations and send CreateFileNode event.
+ * @param {String} name - Name of the created file.
+ * @param {Boolean} isFolder - True if the created file is a folder, otherwise false.
+ * @param {String} path - Path (id) of the created file.
+ * @return {Promise<void>} Promise with nothing on success otherwise an error.
+ */
+async function onCreateFile({ name, isFolder, path }) {
+  const parentNodePath = path.substring(0, path.lastIndexOf('/')) || props.projectName;
+
+  if (isFolder) {
+    // Make empty folder visible by the FileExplorer.
+    // TODO: Refacto when FileInformation have isFolder property.
+    localFileInformations.value.push(new FileInformation({ path: `${path}/__empty__` }));
+    FileEvent.CreateFileNodeEvent.next({ parentNodePath, node: null, isFolder });
+  } else {
+    await getStatus(props.projectName, [path], (filePath) => filePath === path)
+      .then(([fileStatus]) => {
+        localFileInformations.value.push(fileStatus);
+
+        FileEvent.CreateFileNodeEvent.next({
+          parentNodePath,
+          node: { id: path, label: name },
+          isFolder,
+        });
+      });
   }
 }
 
 /**
- * Get projects files and their status, then call function to make the update.
+ * Remove deleted file from localFileInformations.
+ * @param {Object} file - The deleted file.
  */
-function updateProjectFiles() {
-  return Promise.allSettled([
-    getProjectFiles(props.projectName).then((fileInformations) => {
-      updateFileExplorer(fileInformations);
-      return fileInformations;
-    }),
-    getStatus(props.projectName),
-  ]).then((allResults) => {
-    const filesStatus = allResults[1].value;
-    const fileInformations = allResults[0].value.map((file) => {
-      const statusResult = filesStatus.find(({ path }) => file.path === path);
-
-      if (statusResult) {
-        return statusResult;
-      }
-      return new FileStatus({ path: file.path });
-    });
+function onDeleteFile(file) {
+  const temp = localFileInformations.value
```

Comment:
can you rename `temp` to be more readeable ?

Reply:
done